packages/accounts/src/msca/account.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  BaseSmartContractAccount,
+  SignerSchema,
+  createBaseSmartAccountParamsSchema,
+  type BatchUserOperationCallData,
+  type SignTypedDataParams,
+  type SmartAccountSigner,
+  type SupportedTransports,
+} from "@alchemy/aa-core";
+import {
+  concatHex,
+  encodeFunctionData,
+  type Address,
+  type FallbackTransport,
+  type Hex,
+  type Transport,
+} from "viem";
+import { z } from "zod";
+import { IStandardExecutorAbi } from "./abis/IStandardExecutor.js";
+import { MultiOwnerMSCAFactoryAbi } from "./abis/MultiOwnerMSCAFactory.js";
+
+export const createModularSmartContractAccountSchema = <
+  TTransport extends SupportedTransports = Transport
+>() =>
+  createBaseSmartAccountParamsSchema<TTransport>().extend({
+    owner: SignerSchema,
+    index: z.bigint().optional().default(0n),
+  });
+
+export type ModularSmartContractAccountParams = z.input<
+  ReturnType<typeof createModularSmartContractAccountSchema>
+>;
+
+export class ModularSmartContractAccount<
+  TTransport extends Transport | FallbackTransport = Transport
+> extends BaseSmartContractAccount<TTransport> {
+  protected owner: SmartAccountSigner;
+  protected index: bigint;
+
+  constructor(params_: ModularSmartContractAccountParams) {
+    const params =
+      createModularSmartContractAccountSchema<TTransport>().parse(params_);
+
+    super(params);
+
+    this.owner = params.owner;
+    this.index = params.index;
+  }
+
+  getDummySignature(): Hex {
+    throw new Error("Method not implemented.");
```

Comment:
for MSCA, I think this should actually be defined on the signer itself... which raises a question for @avasisht23 
Should we add `getDummySignature` as a method on the SmartAccountSigner impls? we could make it an optional lambda property for now or something and intro a breaking change later

but the dummy signature is something that is dependent on the type of signature validation the account does